Segregator.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace ConsoleApp2
+{
+    class Segregator:logics
+    {
+        
+        public string find_data(List<int> data)
+        {
+            int j = 0;
+            string ret_data=string.Empty;
+            data.Sort();
+            var val = find_value(data);
+
+            foreach (var kk in val)
+            {
+                ret_data += "{" + kk.Key[0].ToString() + "-" + kk.Key[1] + ", " + kk.Value.ToString()+"} ";
+                j++;
+            }
+
+            return ret_data;
+        }
+
+        public Dictionary<List<int>,int> find_value(List<int> arr)
+        {
+            var val = new Dictionary<List<int>, int>();
```

Comment:
Usage of dictionary is not for right purpose. Even though this works, Dictionary is  data structure meant for fast lookup with a key and value (value could be any complex data structure).
Right approach would be to define a list of tuple or a list of simple structure or class with start, end and count as members.

Reply:
is this comment addressed?